wqflask/wqflask/correlation/pre_computes.py

```diff
@@ -156,3 +160,39 @@ def get_datasets_data(base_dataset, target_dataset_data):
         samples_fetched, base_traits_data)
 
     return (target_results, base_results)
+
+
+def get_file_size(file_path):
+    """params:file_path
+       return:size of file in MB
+"""
+    file = Path(file_path)
+    size_in_bytes = file.stat().st_size
+    return float(size_in_bytes)/(1024**2)
+
+
+def consume_json_file_handler(file_path, prefix=""):
+    """function to read any json file stream
+    or load based on size assumption:one object"""
+
+    with open(file_path) as file_handler:
+        if get_file_size(file_path) > 50:
+
+            streamed_json = [obj for obj in ijson.items(
+                file_handler, prefix, use_float=True)]
```

Comment:
*issue: question:* shouldn't we use an iterator for this as well?  We use ijson, we is an iterator, and then convert it into a list, which beats the point of using ijson in the first place.